src/giskardpy/tree_manager.py

```diff
@@ -0,0 +1,140 @@
+from giskardpy.plugin import PluginBehavior
+from giskardpy import logging
+from sortedcontainers import SortedList
+
+class TreeManager():
+
+    class ManagerNode():
+        def __init__(self, node, parent, position):
```

Comment:
please add doc strings with type hints and a description, if the functionality is not obvious, everywhere in this file.